controller/api/public/top_routes.go

```diff
@@ -355,8 +355,6 @@ func processRouteMetrics(results []promResult, timeWindow string, table indexedT
 			}
 		}
 	}
-	if samples == 0 {
-		return errors.New("No samples")
-	}
+
```

Comment:
now that we're not checking `samples`, you can just remove the variable altogether. and going a step further, this function can no longer return an error, so remove the return type.